<?xml version="1.0"?>
<launch>
    <!-- declare arg to be passed in -->
    <arg name="team_side" default="home" />
    <arg name="sim_mode" default="False" />
    <arg name="two_robots" default="false" />

    <!-- Outputs commands to either real world robot or simulation robot-->
<!--     <arg name="vel_cmd_ally1" default="vel_cmds" />
    <arg name="vel_cmd_ally2" default="vel_cmds" />
    <arg name="kick_ally1" default="kick" />
    <arg name="kick_ally2" default="kick" /> -->

    <group ns="provospaceprogram_home">

        <!-- Set ROS parameters that the code can access -->
        <param name="team_side" type="string" value="$(arg team_side)"/>

        <node name="vision" pkg="provospaceprogram" type="vision.py" output="screen">
            <remap from="game" to="/game_state" />
            <param name="is_team_home" value="$(eval team_side == 'home')" />
        </node>

        <node name="ball_estimator_node" pkg="provospaceprogram" type="estimatornode.py" output="screen">
            <remap from="game" to="/game_state" />
            <remap from="subscriber" to="ball"/>
            <remap from="publisher" to="ball_estimator"/>
            <param name="is_team_home" value="$(eval team_side == 'home')" type="bool"/>
        </node>

        <node name="ally1_estimator_node" pkg="provospaceprogram" type="estimatornode.py" output="screen">
            <remap from="game" to="/game_state" />
            <remap from="subscriber" to="ally1"/>
            <remap from="publisher" to="ally1_estimator"/>
            <param name="is_team_home" value="$(eval team_side == 'home')" type="bool"/>
        </node>

        <node name="ally2_estimator_node" pkg="provospaceprogram" type="estimatornode.py" output="screen">
            <remap from="game" to="/game_state" />
            <remap from="subscriber" to="ally2"/>
            <remap from="publisher" to="ally2_estimator"/>
            <param name="is_team_home" value="$(eval team_side == 'home')" type="bool"/>
        </node>

        <node name="ai_node1" pkg="provospaceprogram" type="ai_node.py" output="screen">
            <remap from="me" to="ally1_estimator" />
            <remap from="ally" to="ally2_estimator" />
            <remap from="opp1" to="opp1_estimator" />
            <remap from="opp2" to="opp2_estimator" />
            <remap from="game" to="/game_state" />
            <param name="is_team_home" value="$(eval team_side == 'home')" />
            <param name="is_player1" value="True" type="bool"/>
        </node>

        <node name="ai_node2" pkg="provospaceprogram" type="ai_node.py" output="screen">
            <remap from="me" to="ally2_estimator" />
            <remap from="ally" to="ally1_estimator" />
            <remap from="opp1" to="opp1_estimator" />
            <remap from="opp2" to="opp2_estimator" />
            <remap from="game" to="/game_state" />
            <param name="is_team_home" value="$(eval team_side == 'home')" />
            <param name="is_player1" value="False" type="bool"/>
        </node>

    </group>
</launch>
